src/rafimage.cpp

```diff
@@ -296,6 +292,37 @@ namespace Exiv2 {
         io_->read(jpg_img_length, 4);
         long jpg_img_off = Exiv2::getULong((const byte *) jpg_img_offset, bigEndian);
         long jpg_img_len = Exiv2::getULong((const byte *) jpg_img_length, bigEndian);
+        byte cfa_header_offset [4];
+        io_->read(cfa_header_offset, 4);
+        // byte cfa_header_length [4];
+        // io_->read(cfa_header_length, 4);
+        long cfa_hdr_off = Exiv2::getULong((const byte *) cfa_header_offset, bigEndian);
+        // long cfa_hdr_len = Exiv2::getULong((const byte *) cfa_header_length, bigEndian);
+        io_->seek(cfa_hdr_off,BasicIo::beg);
+        byte byte_count[4];
+        io_->read(byte_count, 4);
+        byte byte_tag[2];
+        byte byte_size[2];
+        int32_t count = getLong(byte_count, bigEndian);
+        for (int32_t i = 0; i < count; ++i ) {
```

Comment:
I wonder if this loop here could affect to the call `TiffParser::decode` we have later in line 307 :thinking:

Reply:
Unlikely, as `TiffParser::decode` shares no internal state with `this` and all the parameters appear to be untouched by the new code.

Reply:
Unlikely, as `TiffParser::decode` shares no internal state with `this` and all the parameters appear to be untouched by the new code.